recipes-bsp/u-boot/files/0011-board-siemens-iot2050-get-and-set-m.2-releated-infor.patch

```diff
@@ -0,0 +1,229 @@
++	{M2_B_KEY, {0,1,0,0}, {0,0,1}, {PCIE0_LANE0, PCIE0_LANE1}},
++	{M2_B_KEY, {0,0,0,1}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{M2_B_KEY, {0,1,1,0}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{M2_B_KEY, {1,0,0,1}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{M2_B_KEY, {1,1,0,1}, {0,1,0}, {PCIE0_LANE0, PCIE1_LANE0}},
++	{M2_B_KEY, {0,0,0,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{M2_B_KEY, {0,1,0,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{M2_B_KEY, {0,0,1,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++	{M2_B_KEY, {1,0,1,1}, {1,1,0}, {USB30, PCIE1_LANE0}},
++};
++
++static gpioConfig_t m2_pinmux_config[PCIE_MUX_CONTROL] = {
++	{"SEL_USB30_PCIE0_LANE0", "gpio@600000_88", GPIOD_IS_OUT},
++	{"SEL_PCIE1_PCIE0",       "gpio@600000_82", GPIOD_IS_OUT},
++	{"SEL_USB30_PCIE0_LANE1", "gpio@600000_89", GPIOD_IS_OUT},
++};
++
++static gpioConfig_t m2_key_config[M2_CONFIG_LEN] = {
++	{"KEY_CONFIG_3", "gpio@601000_89", GPIOD_IS_IN},
++	{"KEY_CONFIG_2", "gpio@601000_88", GPIOD_IS_IN},
++	{"KEY_CONFIG_1", "gpio@601000_19", GPIOD_IS_IN},
++	{"KEY_CONFIG_0", "gpio@601000_18", GPIOD_IS_IN},
++};
++
++static gpioConfig_t m2_enable_power = {"P3V3_M2_EN", "gpio@601000_17", GPIOD_IS_OUT};
++
++static int Get_PinValue(const char *gpio_name,const char *label)
++{
++	struct gpio_desc gpio;
++
++	if (dm_gpio_lookup_name(gpio_name, &gpio) < 0 ||
++	dm_gpio_request(&gpio, label) < 0 ||
++	dm_gpio_set_dir_flags(&gpio, GPIOD_IS_IN) < 0)
++		return false;
++
++	return dm_gpio_get_value(&gpio);
++}
++
++static void Set_PinValue(const char *gpio_name,const char *label,int value)
++{
++	struct gpio_desc gpio;
++
++	if (dm_gpio_lookup_name(gpio_name, &gpio) < 0 ||
++	dm_gpio_request(&gpio, label) < 0 ||
++	dm_gpio_set_dir_flags(&gpio, GPIOD_IS_OUT) < 0) {
++		pr_err("IOT2050:Can not set the pin value\n");
++		return ;
++	}
++	dm_gpio_set_value(&gpio,value);
++}
++
++static int *Get_M2_Config(void)
++{
++	int i;
++	static int result[M2_CONFIG_LEN] = {0};
++	for(i=0; i < M2_CONFIG_LEN; i++)
++	{
++		result[i] = Get_PinValue(m2_key_config[i].gpio_name, m2_key_config[i].label);
++	}
++	return result;
++}
++
++static serdes_config_t Get_Serdes_Config_Set_M2_Pinmux(void)
++{
++	int i,j;
++	serdes_config_t serdes_config = {USB30, PCIE1_LANE0};//default
++	int config_table_size= sizeof(config_table)/sizeof(config_table[0]);
++
++    //1.TBD:check eeprom or flash
++    //2.configure the m2
++	int *config = Get_M2_Config();
++	printf("M2-KEY PIN:");
++	for(i=0; i < M2_CONFIG_LEN;i++)
++	{
++		printf("%d",*(config+i));
++	}
++	printf("\n");
++
++	for(i = 0; i < config_table_size; i++)
++	{
++		if(!memcmp(config, config_table[i].m2config, M2_CONFIG_LEN * sizeof(int)))
++		{
++			// set m2 pinmux
++			for(j = 0; j < PCIE_MUX_CONTROL; j++)
++				Set_PinValue(m2_pinmux_config[j].gpio_name, m2_pinmux_config[j].label, config_table[i].m2_pinmux_value[j]);
++			//get serdes config info
++			serdes_config =  config_table[i].serdes_config;
++			break;
++		}
++	}
++
++	return serdes_config;
++}
```

Comment:
This function is actually doing two things:

1. set the m.2 pcie mux
2. get the serdes info, so that later the dtbo could be determined accordingly.

So why not split into two functions?